### PR TITLE
feat(dirty-check-plugin) add isPathDirty api

### PR DESCRIPTION
**Pull request by ritox842**
Add api for dirty-check and entity-dirty-check plugins for 'isPathDirty'.
By giving the method path (For ex. 'user.location.city') you can know if this specific value was changed (from last head value).

### CODE REVIEW

File: akita/__tests__/dirty-check.spec.ts [hunk 1]
```diff
@@ -135,6 +135,35 @@ describe('DirtyCheck', () => {
         const hasHead = dirtyCheck.hasHead();
         expect(hasHead).toBeFalsy();
       });
+
+      describe('dirtyPath',()=>{
+        const widgetsStore = new WidgetsStore({ some: { nested: { value: '' } } });
+        const widgetsQuery = new WidgetsQuery(widgetsStore);
+        const dirtyCheck = new DirtyCheckPlugin(widgetsQuery).setHead();
+        it('should not return dirty for isPathDirty', function() {
```
Comment (NetanelBasal) [inline akita/__tests__/dirty-check.spec.ts]: The tests are not cover the whole process.

File: akita/__tests__/dirty-check.spec.ts [hunk 2]
```diff
@@ -135,6 +135,35 @@ describe('DirtyCheck', () => {
...
+          dirtyCheck.setHead();
+          const firstWidget = createWidget();
+          widgetsStore.add(firstWidget);
+          isPathDirty = dirtyCheck.isPathDirty('some.nested.value');
+          expect(isPathDirty).toBeFalsy();
+        });
+
+        it('should return dirty for isPathDirty', function() {
+          let isPathDirty: boolean;
+          dirtyCheck.setHead();
```
Comment (NetanelBasal) [inline akita/__tests__/dirty-check.spec.ts]: I'm not sure why you are calling `setHead` again.

File: akita/__tests__/dirty-check.spec.ts [hunk 3]
```diff
@@ -290,6 +319,32 @@ describe('DirtyCheckEntity', () => {
       widgetsStore.add(widget);
       expect(collection.hasHead(widget.id)).toBeTruthy();
     });
+
+    describe('dirtyPath',()=>{
+      it('should not return dirty for isPathDirty', function() {
```
Comment (NetanelBasal) [inline akita/__tests__/dirty-check.spec.ts]: same here, the whole process.

File: akita/src/plugins/dirty-check/dirty-check-plugin.ts [hunk 4]
```diff
@@ -18,6 +18,10 @@ export const dirtyCheckDefaultParams = {
   comparator: (head, current) => JSON.stringify(head) !== JSON.stringify(current)
 };
 
+export function getNestedObject (nestedObj, pathArr) {
```
Comment (NetanelBasal) [inline akita/src/plugins/dirty-check/dirty-check-plugin.ts]: getNestedObject => getNestedPath

Comment (NetanelBasal) [inline akita/src/plugins/dirty-check/dirty-check-plugin.ts]: types

File: akita/src/plugins/dirty-check/dirty-check-plugin.ts [hunk 5]
```diff
@@ -125,6 +132,20 @@ export class DirtyCheckPlugin<Entity = any, StoreState = any> extends AkitaPlugi
     return head;
   }
 
+  isPathDirty(path: string) {
+      const head = this.getHead();
+      const current = (this.getQuery() as Query<StoreState>).getSnapshot();
+      const pathAsArray: string[] = path.split('.');
+      try {
```
Comment (NetanelBasal) [inline akita/src/plugins/dirty-check/dirty-check-plugin.ts]: Why do we need try and catch block?

File: akita/src/plugins/dirty-check/dirty-check-plugin.ts [hunk 6]
```diff
@@ -125,6 +132,20 @@ export class DirtyCheckPlugin<Entity = any, StoreState = any> extends AkitaPlugi
     return head;
   }
 
+  isPathDirty(path: string) {
+      const head = this.getHead();
+      const current = (this.getQuery() as Query<StoreState>).getSnapshot();
+      const pathAsArray: string[] = path.split('.');
```
Comment (NetanelBasal) [inline akita/src/plugins/dirty-check/dirty-check-plugin.ts]: You should make the splitting inside the function.

File: akita/src/plugins/dirty-check/entity-dirty-check-plugin.ts [hunk 7]
```diff
@@ -67,7 +74,23 @@ export class EntityDirtyCheckPlugin<E, P extends DirtyCheckPlugin<E, any> = Dirt
     return this.checkSomeDirty(entities);
   }
 
-  someDirty = this.isSomeDirty;
+  isPathDirty(id: ID, path: string) {
+    if (this.entities.has(id)) {
+      const head = (this.getEntity(id) as any).getHead();
```
Comment (NetanelBasal) [inline akita/src/plugins/dirty-check/entity-dirty-check-plugin.ts]: You are not DRY.

File: akita/__tests__/dirty-check.spec.ts (hunk 1 above)
Comment (ritox842) [reply akita/__tests__/dirty-check.spec.ts]: Why? Iv'e added a test for changed value and unchanged value in both entity and store. What else should I cover?

Comment (NetanelBasal) [reply akita/__tests__/dirty-check.spec.ts]: another cycle

**Pull request closed without merging**